src/mssql_server/mod.rs

```diff
@@ -0,0 +1,187 @@
+/// use testcontainers_modules::mssql_server;
+///
+/// let docker = clients::Cli::default();
+/// let mssql_server = docker.run(mssql_server::MssqlServer::default());
+/// let ado_connection_string = format!(
+///    "Server=tcp:127.0.0.1,{};Database=test;User Id=sa;Password=yourStrong(!)Password;TrustServerCertificate=True;",
+///    mssql_server.get_host_port_ipv4(1433)
+/// );
+/// ```
+///
+/// # Environment variables
+///
+/// Refer to the [documentation](https://learn.microsoft.com/en-us/sql/linux/sql-server-linux-configure-environment-variables)
+/// for a complite list of environment variables.
+///
+/// Following environment variables are required.
+/// A image provided by this module has default values for them.
+///
+/// ## `ACCEPT_EULA`
+///
+/// You need to accept the [End-User Licensing Agreement](https://go.microsoft.com/fwlink/?linkid=857698)
+/// before using the SQL Server image provided by this module.
+/// To accept EULA, you can set this environment variable to `Y`.
+/// The default value is `Y`.
+///
+/// ## `MSSQL_SA_PASSWORD`
+///
+/// The SA user password. This password is required to conform to the
+/// [strong password policy](https://learn.microsoft.com/en-us/sql/relational-databases/security/password-policy?view=sql-server-ver15#password-complexity).
+/// The default value is `yourStrong(!)Password`.
+///
+/// ## `MSSQL_PID`
+///
+/// The edition of SQL Server.
+/// The default value is `Developer`, which will run the container using the Developer Edition.
+#[derive(Debug)]
+pub struct MssqlServer {
+    env_vars: HashMap<String, String>,
+}
+
+impl MssqlServer {
+    const NAME: &'static str = "mcr.microsoft.com/mssql/server";
+    const TAG: &'static str = "2022-CU10-ubuntu-22.04";
+    const DEFAULT_SA_PASSWORD: &'static str = "yourStrong(!)Password";
+
+    /// Sets the password as `MSSQL_SA_PASSWORD`.
+    pub fn with_sa_password(self, password: impl Into<String>) -> Self {
+        let mut env_vars = self.env_vars;
+        env_vars.insert("MSSQL_SA_PASSWORD".to_owned(), password.into());
+
+        Self { env_vars }
+    }
+}
+
+impl Default for MssqlServer {
+    fn default() -> Self {
+        let mut env_vars = HashMap::new();
+        env_vars.insert("ACCEPT_EULA".to_owned(), "Y".to_owned());
+        env_vars.insert(
+            "MSSQL_SA_PASSWORD".to_owned(),
+            Self::DEFAULT_SA_PASSWORD.to_owned(),
+        );
+        env_vars.insert("MSSQL_PID".to_owned(), "Developer".to_owned());
+
+        Self { env_vars }
+    }
+}
+
+impl Image for MssqlServer {
+    type Args = ();
+
+    fn name(&self) -> String {
+        Self::NAME.to_owned()
+    }
+
+    fn tag(&self) -> String {
+        Self::TAG.to_owned()
+    }
+
+    fn ready_conditions(&self) -> Vec<WaitFor> {
+        // Wait until the server is ready for connections and all system databases are recovered
+        vec![WaitFor::StdOutMessage {
+            message: "Recovery is complete".to_owned(),
+        }]
```

Comment:
One condition is enough? Shouldn't we have something like `ready to accept connections`?

Reply:
Actually "SQL Server is now ready for client connections" (deleted the check in 1637d52) also appears before "Recovery is complete", it makes more sense to check both messages. I will update the condition.

Reply:
Thanks a lot!